qt-models/divetripmodel.cpp

```diff
@@ -717,7 +717,7 @@ void DiveTripModelTree::populate()
 	for (int i = 0; i < dive_table.nr; ++i) {
 		dive *d = get_dive(i);
 		update_cylinder_related_info(d);
```

Comment:
If `d` were NULL, wouldn't `update_cylinder_related_info()` crash? I.e. shouldn't the null-pointer check be moved in front of this calls? As it is, it makes no sense...

Reply:
`update_cylinder_related_info()` does its own NULL check...
This way it was less of a code change. But for clarity, year, you're right.